pkg/icingadb/db.go

```diff
@@ -342,6 +365,49 @@ func (db *DB) NamedBulkExec(
 	return g.Wait()
 }
 
+// NamedExecContextBulk performs db.NamedExecContext(ctx, query, rows).
+// If that fails due to upserting the same row twice (Postgres only), it splits rows and upserts them in a transaction.
+func (db *DB) NamedExecContextBulk(ctx context.Context, query string, rows []contracts.Entity) error {
```

Comment:
Is that function even used anywhere? `git grep NamedExecContextBulk` brings up no other occurrences than these two right here.